packages/application-shell/src/components/application-shell-provider/application-shell-provider.spec.js

```diff
@@ -32,6 +31,10 @@ const createTestProps = props => ({
 });
 
 describe('rendering', () => {
+  beforeEach(() => {
+    // eslint-disable-next-line global-require
+    require('../authenticated').setIsAuthenticated('true');
```

Comment:
Q: Can't we just import this as `import { setIsAuthenticated } from '../authenticated'`? I presume, from reading the code, you ran into issues with mocking in L11. I find this `require`-cjs work around a bit confusing. Would a 

```js
jest.mock('../authenticated', () => ({
  ...require.requireActual('../authenticated'),
  // eslint-disable-next-line react/prop-types
  Authenticated: ({ children }) => <div>{children}</div>,
}));
```

Might be a bit trouble some with the default export of `Authenticated` and I am not 100% sure what the issue here is but just wanted to share the thouhgt.

Reply:
I've spent a bit of time on this actually.

The main problem is that the `ApplicationShellProvider` renders the normal apollo provider. So if we test it and wrap it with an apollo mock provider to mock the graphql request, it won't work because the normal apollo provider takes precedence over the mocked one. This is only a "problem" with this component, as it's the top level one.

So the only solution is to mock the `Authenticated` component using a manual mock, and we need to be able to control the `isAuthenticated` value from outside of the mock.

I haven't tried with `import { setIsAuthenticated } from '../authenticated'`, will do that tomorrow. Thanks!